commit 99e541f257fd
Author: Michael Kaply <mozilla@kaply.com>
Date:   Wed Jun 6 15:58:24 2018 -0500

    Bug 1466863 - Just use empty metadata if invalid. r=florian, a=RyanVM
    
    MozReview-Commit-ID: 30Q5Sdi5ZRt
    
    --HG--
    extra : source : 5404016c6d164a29c8dccdce58363d5b893af565
---
 toolkit/components/search/nsSearchService.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git toolkit/components/search/nsSearchService.js toolkit/components/search/nsSearchService.js
index 86a286819f02b..6af0f3e64f39a 100644
--- toolkit/components/search/nsSearchService.js
+++ toolkit/components/search/nsSearchService.js
@@ -2074,7 +2074,7 @@ Engine.prototype = {
   },
 
   getAttr(name) {
-    return (this._metaData && this._metaData[name]) || undefined;
+    return this._metaData[name] || undefined;
   },
 
   // nsISearchEngine
@@ -3191,7 +3191,7 @@ SearchService.prototype = {
       let name = engine._name;
       if (name in this._engines) {
         LOG("_loadEnginesMetadataFromCache, transfering metadata for " + name);
-        this._engines[name]._metaData = engine._metaData;
+        this._engines[name]._metaData = engine._metaData || {};
       }
     }
   },
